src/go/rpk/pkg/config/weak.go

```diff
@@ -0,0 +1,592 @@
+	c.Redpanda = internal.Redpanda
+	c.Rpk = internal.Rpk
+	c.Pandaproxy = internal.Pandaproxy
+	c.PandaproxyClient = internal.PandaproxyClient
+	c.SchemaRegistry = internal.SchemaRegistry
+	c.SchemaRegistryClient = internal.SchemaRegistryClient
+	c.Other = internal.Other
+
+	return nil
+}
+
+func (rpc *RedpandaConfig) UnmashalYAML(n *yaml.Node) error {
+	var internal struct {
+		Directory                  weakString             `yaml:"data_directory"`
+		ID                         weakInt                `yaml:"node_id" `
+		Rack                       weakString             `yaml:"rack,omitempty"`
+		SeedServers                seedServers            `yaml:"seed_servers"`
+		RPCServer                  SocketAddress          `yaml:"rpc_server"`
+		RPCServerTLS               serverTLSArray         `yaml:"rpc_server_tls,omitempty"`
+		KafkaAPI                   namedSocketAddresses   `yaml:"kafka_api"`
+		KafkaAPITLS                serverTLSArray         `yaml:"kafka_api_tls,omitempty"`
+		AdminAPI                   namedSocketAddresses   `yaml:"admin"`
+		AdminAPITLS                serverTLSArray         `yaml:"admin_api_tls,omitempty"`
+		CoprocSupervisorServer     SocketAddress          `yaml:"coproc_supervisor_server,omitempty"`
+		AdminAPIDocDir             weakString             `yaml:"admin_api_doc_dir,omitempty"`
+		DashboardDir               weakString             `yaml:"dashboard_dir,omitempty"`
+		CloudStorageCacheDirectory weakString             `yaml:"cloud_storage_cache_directory,omitempty"`
+		AdvertisedRPCAPI           *SocketAddress         `yaml:"advertised_rpc_api,omitempty"`
+		AdvertisedKafkaAPI         namedSocketAddresses   `yaml:"advertised_kafka_api,omitempty"`
+		DeveloperMode              weakBool               `yaml:"developer_mode"`
+		Other                      map[string]interface{} `yaml:",inline"`
+	}
+	if err := n.Decode(&internal); err != nil {
+		return err
+	}
+	rpc.Directory = string(internal.Directory)
+	rpc.ID = int(internal.ID)
+	rpc.Rack = string(internal.Rack)
+	rpc.SeedServers = internal.SeedServers
+	rpc.RPCServer = internal.RPCServer
+	rpc.RPCServerTLS = internal.RPCServerTLS
+	rpc.KafkaAPI = internal.KafkaAPI
+	rpc.KafkaAPITLS = internal.KafkaAPITLS
+	rpc.AdminAPI = internal.AdminAPI
+	rpc.AdminAPITLS = internal.AdminAPITLS
+	rpc.CoprocSupervisorServer = internal.CoprocSupervisorServer
+	rpc.AdminAPIDocDir = string(internal.AdminAPIDocDir)
+	rpc.DashboardDir = string(internal.DashboardDir)
+	rpc.CloudStorageCacheDirectory = string(internal.CloudStorageCacheDirectory)
+	rpc.AdvertisedRPCAPI = internal.AdvertisedRPCAPI
+	rpc.AdvertisedKafkaAPI = internal.AdvertisedKafkaAPI
+	rpc.DeveloperMode = bool(internal.DeveloperMode)
+	rpc.Other = internal.Other
+	return nil
+}
+
+func (rpkc *RpkConfig) UnmarshalYAML(n *yaml.Node) error {
+	var internal struct {
+		// Deprecated 2021-07-1
+		TLS *TLS `yaml:"tls,omitempty"`
+		// Deprecated 2021-07-1
+		SASL *SASL `yaml:"sasl,omitempty"`
+
```

Comment:
This function looks like a great opportunity to warn about these deprecations as well, and mention that specifying the Kafka TLS and Kafka SASL fields in the rpk config section will be deprecated in 2023?